src/main/java/com/tileman/TilemanModePlugin.java

```diff
@@ -102,11 +104,38 @@ TilemanModeConfig provideConfig(ConfigManager configManager) {
         return configManager.getConfig(TilemanModeConfig.class);
     }
 
+    private final MovementFlag[] fullBlock = new MovementFlag[]
+            {MovementFlag.BLOCK_MOVEMENT_FLOOR,
+                    MovementFlag.BLOCK_MOVEMENT_FLOOR_DECORATION,
+                    MovementFlag.BLOCK_MOVEMENT_OBJECT,
+                    MovementFlag.BLOCK_MOVEMENT_FULL};
+
+    private final MovementFlag[] directionalNorth = new MovementFlag[]
+            {MovementFlag.BLOCK_MOVEMENT_NORTH_WEST,
+                    MovementFlag.BLOCK_MOVEMENT_NORTH,
+                    MovementFlag.BLOCK_MOVEMENT_NORTH_EAST};
+
+    private final MovementFlag[] directionalSouth = new MovementFlag[]
+            {MovementFlag.BLOCK_MOVEMENT_SOUTH_WEST,
+                    MovementFlag.BLOCK_MOVEMENT_SOUTH,
+                    MovementFlag.BLOCK_MOVEMENT_SOUTH_EAST};
+
+    private final MovementFlag[] directionalWest = new MovementFlag[]
+            {MovementFlag.BLOCK_MOVEMENT_NORTH_WEST,
+                    MovementFlag.BLOCK_MOVEMENT_WEST,
+                    MovementFlag.BLOCK_MOVEMENT_SOUTH_WEST};
+
+    private final MovementFlag[] directionalEast = new MovementFlag[]
+            {MovementFlag.BLOCK_MOVEMENT_NORTH_EAST,
+                    MovementFlag.BLOCK_MOVEMENT_EAST,
+                    MovementFlag.BLOCK_MOVEMENT_SOUTH_EAST};
+
+    private final MovementFlag[] allDirections = Stream.of(directionalNorth, directionalSouth, directionalWest, directionalEast).flatMap(Stream::of).toArray(MovementFlag[]::new);
```

Comment:
This will duplicate every corner twice in the allDirections list.

Reply:
Good point :)